gluonnlp/vocab.py

```diff
@@ -456,3 +463,126 @@ def from_json(json_str):
         vocab._bos_token = vocab_dict.get('bos_token')
         vocab._eos_token = vocab_dict.get('eos_token')
         return vocab
+
+class SortedVocab(Vocab):
```

Comment:
Do we necessarily need this class? We probably can make the ``sort`` a transformer for the vocab.

```
>>> x = nlp.data.Counter({'a': 10, 'b': 1, 'c': 2, '<unk>': 3, '<bos>': 1})
>>> y = nlp.data.Counter(x.most_common())
>>> vocab = nlp.Vocab(counter=y)
>>> vocab['a']
0

```

Reply:
The problem with the current Vocab is that special / reserved words are always indexed ahead of common words and unk always indexes to 0. SortedVocab breaks all that assumption

Reply:
I got that. But what can be the negative impact of that since you know the index of the special tokens? If you can just sort the vocab, using sort as a transform, why not using it?